.devcontainer/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM mcr.microsoft.com/vscode/devcontainers/miniconda:0-3
+
+#  I am suprised this is needed
+RUN conda init
+
+# Copy environment.yml (if found) to a temp location so we update the environment. Also
+# copy "noop.txt" so the COPY instruction does not fail if no environment.yml exists.
+COPY .devcontainer/cuda/environment.yml .devcontainer/noop.txt /tmp/conda-tmp/
+RUN if [ -f "/tmp/conda-tmp/environment.yml" ]; then umask 0002 && /opt/conda/bin/conda env update -n base -f /tmp/conda-tmp/environment.yml; fi \
+    && sudo rm -rf /tmp/conda-tmp
+
+# Install the full llvm sweet
+# Tools needed for llvm
+RUN sudo apt-get -y update
+RUN sudo apt install -y lsb-release wget software-properties-common gnupg
+
+# # If USE_CLANG is set we will install
+# Can't build succesfully, The exports need to go in the .sh script
+# But also clang-15 might be two new for cuda 11.8
+
+# ARG USE_CLANG
+# RUN if [ -n "$USE_CLANG" ]; then \
+#     sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"; \
+#     sudo .devcontainer/scripts/update_alternatives_clang.sh 15 10; \
+#     export CC=clang; \
+#     export CXX=clang++; \
+#     fi
```

Comment:
Let's not merge commented out code

Reply:
I have tested this workflow I can build with clang succesfully on CPU but when trying to compile with CUDA: 
```

  #$ "/usr/bin"/clang -D__CUDA_ARCH__=520 -D__CUDA_ARCH_LIST__=520 -E -x c++
  -DCUDA_DOUBLE_MATH_FUNCTIONS -D__CUDACC__ -D__NVCC__
  "-I/opt/conda/bin/../include" -D__CUDACC_VER_MAJOR__=11
  -D__CUDACC_VER_MINOR__=8 -D__CUDACC_VER_BUILD__=89
  -D__CUDA_API_VER_MAJOR__=11 -D__CUDA_API_VER_MINOR__=8
  -D__NVCC_DIAG_PRAGMA_SUPPORT__=1 -include "cuda_runtime.h" -m64
  "CMakeCUDACompilerId.cu" -o "tmp/CMakeCUDACompilerId.cpp1.ii"

  /usr/bin/clang: /opt/conda/bin/../lib/libtinfo.so.6: no version information
  available (required by /usr/lib/x86_64-linux-gnu/libLLVM-11.so.1)

  /usr/bin/clang: /opt/conda/bin/../lib/libtinfo.so.6: no version information
  available (required by /usr/lib/x86_64-linux-gnu/libedit.so.2)

  /usr/bin/clang: symbol lookup error:
  /usr/lib/x86_64-linux-gnu/libLLVM-11.so.1: undefined symbol:
  ffi_type_sint32, version LIBFFI_BASE_7.0

  # --error 0x7f --

  

  

Call Stack (most recent call first):
  /opt/conda/share/cmake-3.22/Modules/CMakeDetermineCompilerId.cmake:6 (CMAKE_DETERMINE_COMPILER_ID_BUILD)
  /opt/conda/share/cmake-3.22/Modules/CMakeDetermineCompilerId.cmake:48 (__determine_compiler_id_test)
  /opt/conda/share/cmake-3.22/Modules/CMakeDetermineCUDACompiler.cmake:298 (CMAKE_DETERMINE_COMPILER_ID)
  cmake/public/cuda.cmake:47 (enable_language)
  cmake/Dependencies.cmake:43 (include)
  CMakeLists.txt:710 (include)
  ```
  
  I think this has to do with the mismatch between conda install cuda libs.  I can remove this Clang stuff entirely but not sure